packages/vscode-extension/src/commands/ListFiles.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import { window } from 'vscode';
+import { COMMAND, LIST_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, FolderUtils } from '../util/ExtensionUtil';
+
+import BaseAction from './BaseAction';
+import { FolderType } from './FolderType';
+
+const COMMAND_NAME = 'listfiles';
+
+const LIST_FILES_COMMAND = {
+	OPTIONS: {
+		FOLDER: 'folder',
+	},
+};
+
+export default class ListFiles extends BaseAction {
+	constructor() {
+		super(COMMAND_NAME);
+	}
+
+	protected async execute(): Promise<void> {
+		const fileCabinetFolders = await this._getListFolders();
+		const filteredFolders = this.filterDisabledFolders(fileCabinetFolders);
+		if (filteredFolders === undefined) {
+			return;
+		}
+
+		const selectedFolder = await this._selectFolder(filteredFolders);
+
+		if (selectedFolder === undefined) {
+			return;
+		}
+
+		await this._listFiles(selectedFolder);
+	}
+
+	protected async _getListFolders() {
+		const listFoldersPromise = FolderUtils.getFileCabinetFolders(this.executionPath, COMMAND_NAME);
```

Comment:
If I understand this correctly, listFolderPromise contains the list of folder from the FolderUtils.getFileCabinetFolders call. I think we should rename this variable to remove the "Promise" suffix. Maybe something like folderList.

Reply:
listFolderPromise is still a Promise... until we call the "await". This is why I´m putting "Promise" in its name